client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+        source_out = source_in + clip_duration
+
+        # define starting frame for future shot
+        frame_start = (
+            clip_in if workfile_start_frame is None
+            else workfile_start_frame
+        )
+        frame_end = frame_start + (clip_duration - 1)
+
+        return {
+            "frameStart": int(frame_start),
+            "frameEnd": int(frame_end),
+            "clipIn": int(clip_in),
+            "clipOut": int(clip_out),
+            "clipDuration": int(otio_clip.duration().value),
+            "sourceIn": int(source_in),
+            "sourceOut": int(source_out)
+        }
+
+    def _get_allowed_product_type_presets(self, pre_create_data):
+        """Filter out allowed product type presets.
+
+        Args:
+            pre_create_data (dict): precreate attributes inputs
+
+        Returns:
+            list: lit of dict with preset items
+        """
+        return [
+            # return dict with name of preset and add preset dict
+            {"product_name": product_name, **preset}
+            for product_name, preset in self.get_product_presets_with_names().items()  # noqa
+            if pre_create_data[product_name]
+        ]
+
+    def _validate_clip_for_processing(self, otio_clip):
+        """Validate otio clip attributes
+
+        Args:
+            otio_clip (otio.Clip): otio clip object
+
+        Returns:
+            bool: True if all passing conditions
+        """
+        if otio_clip.name is None:
+            return False
+
+        if isinstance(otio_clip, otio.schema.Gap):
+            return False
+
+        # skip all generators like black empty
+        if isinstance(
+            otio_clip.media_reference,
+                otio.schema.GeneratorReference):
+            return False
+
+        # Transitions are ignored, because Clips have the full frame
+        # range.
+        if isinstance(otio_clip, otio.schema.Transition):
+            return False
+
+        return True
+
+    def get_pre_create_attr_defs(self):
+        """Creating pre-create attributes at creator plugin.
+
+        Returns:
+            list: list of attribute object instances
+        """
+        # Use same attributes as for instance attrobites
+        attr_defs = [
+            FileDef(
+                "sequence_filepath_data",
+                folders=False,
+                extensions=[".edl", ".xml", ".aaf", ".fcpxml"],
+                allow_sequences=False,
+                single_item=False,
+                label="Sequence file",
+            ),
+            FileDef(
+                "folder_path_data",
+                folders=True,
+                single_item=False,
+                extensions=[],
+                allow_sequences=False,
+                label="Folder path",
+            ),
+            # TODO: perhaps better would be timecode and fps input
+            NumberDef("timeline_offset", default=0, label="Timeline offset"),
+            UISeparatorDef(),
```

Comment:
Separators should have filled key if there are more than one, otherwise they are joined together.